File: src/kyu_7/deodorant_evaporator/cpp/deodorant_evaporator.cpp
#include <cmath>

using namespace std;
class Evaporator
{

  public:
  static int evaporator(double, double evap_per_day, double threshold);
};

int Evaporator::evaporator(double, double evap_per_day, double threshold) {
  const double evapFactor {(100.0 - evap_per_day) / 100.0};
  const double normalizedThreshold {threshold / 100.0};
  const double daysToThreshold {std::log (normalizedThreshold) / std::log(evapFactor)};
  
  return static_cast <int> (std::ceil(daysToThreshold));
}

File: src/kyu_7/deodorant_evaporator/csharp/deodorant_evaporator.cs
public class Evaporator {

  public static int evaporator(double content, double evap_per_day, double threshold) {
    int day = 0;
    double total = content;
    while (content >= total * (1d / 100) * threshold) {
      content -= content * (1d / 100) * evap_per_day;
      day++;
    }
    return day;
  }
}